core/vnl/vnl_vector.h

```diff
@@ -214,14 +198,65 @@ class VNL_EXPORT vnl_vector
   // Return new vector = -1*(*this)
   vnl_vector<T> operator-() const;
 
-  vnl_vector<T> operator+(T v) const { return vnl_vector<T>(*this, v, vnl_tag_add()); }
-  vnl_vector<T> operator-(T v) const { return vnl_vector<T>(*this, v, vnl_tag_sub()); }
-  vnl_vector<T> operator*(T v) const { return vnl_vector<T>(*this, v, vnl_tag_mul()); }
-  vnl_vector<T> operator/(T v) const { return vnl_vector<T>(*this, v, vnl_tag_div()); }
+  vnl_vector<T> operator+(T v) const {
+    vnl_vector<T> result(this->size());
+    std::transform(this->begin(), this->end(), result.begin(),
+                   [v](T d) -> T { return d + v; });
+    return result;
```

Comment:
`return result` is potentially expensive, because `vnl_vector` does not have a no fast move-constructor. I think it would be better to postpone this change until we have restored fast move-semantics for `vnl_vector`. (Because I think you cannot fully rely on NRVO -- named return value optimization.)

Reply:
Done.